src/backend/api/meals.js

```diff
@@ -1,15 +1,199 @@
+const router = express.Router()
+const knex = require('../database')
 
-router.get("/", async (request, response) => {
+// GET method
+router.get('/', async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    const { maxPrice, availableReservations, title, createdAfter, limit } =
+      request.query
+    let meals = await knex('meal')
+
+    // Get meals price smaller than maxPrice
+    if (maxPrice) {
+      const mealMaxPrice = parseInt(maxPrice)
+      meals = await knex('meal').select('*').where('price', '<', mealMaxPrice)
+    }
+    // Get meals that partially match a title
+    if (title) {
+      meals = await knex('meal')
+        .select('*')
+        .where('title', 'like', `%${title}%`)
+    }
+    //Get meals that has been created after the date
+    if (createdAfter) {
+      const afterDate = new Date(createdAfter)
+      meals = await knex('meal')
+        .select('*')
+        .where('created_date', '>', afterDate)
+    }
+    // Only specific number of meals
+    if (limit) {
+      const mealLimit = parseInt(limit)
+      meals = await knex('meal').select('*').limit(mealLimit)
+    }
+    // Get meals that still has available reservations
+    if (availableReservations) {
+      meals = await knex('meal')
+        .leftJoin('reservation', 'meal.id', '=', 'reservation.meal_id')
+        .select(
+          'meal.id',
+          'meal.title',
+          'meal.description',
+          'meal.location',
+          'meal.when',
+          'meal.created_date',
+          'meal.price',
+          'meal.max_reservations',
+          knex.raw(
+            'COALESCE(SUM(reservation.number_of_guests),0) as total_guests'
+          ),
+          knex.raw(
+            '(meal.max_reservations-COALESCE(SUM(reservation.number_of_guests),0)) AS available_reservation'
+          )
+        )
+        .groupBy('meal.id')
+        .having(
+          knex.raw(
+            '(max_reservations > COALESCE(SUM(reservation.number_of_guests),0))'
+          )
+        )
+        .groupBy('meal.id')
+    }
+    response.json(meals)
+  } catch (error) {
+    throw error
+  }
+})
+
+//GET. Returns meal by id
+router.get('/:id', async (request, response) => {
+  try {
+    const mealId = Number(request.params.id)
+    if (isNaN(mealId)) {
+      response.status(400).json({
+        error: "Meal wasn't found.",
+        hint: 'Id should be a number',
+      })
+      return
+    } else {
+      const meals = await knex('meal')
+      const isFoundId = meals.some((meal) => meal.id === mealId)
+      if (!isFoundId) {
+        response.status(400).json({
+          error: "Meal wasn't found.",
+          hint: 'Meal with this id does not exist',
+        })
+        return
+      }
+      const mealById = await knex('meal').select('*').where({ id: mealId })
+      response.json(mealById)
```

Comment:
Oh, I just noticed you're already doing this here! Nice! So I guess you can remove the else clause above? Or what was your reasoning for it?